### PR TITLE
E2E/Playwright: MM-T5424 Find channel limit to 50 results

### PR DESCRIPTION
**Pull request by saturninoabril**
#### Summary
- Automated  test for [MM-T5424](https://github.com/mattermost/mattermost-test-management/blob/f04ab54c4a2f2d888fe490d29b936d263cd0e80f/data/test-cases/channels/search/MM-T5424.md)
- Added components: `FindChannelsModal`, `ChannelsHeaderMobile`, `ChannelsSidebarLeft`
- Fixed aria-label of "Find Channels" modal

#### Ticket Link
none

#### Screenshots
Screenshot 2023-05-03 at 2 42 21 PM

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Review by lindy65 — APPROVED** [review]
Thanks @saturninoabril - looks good to me 👍

File: webapp/channels/src/components/quick_switch_modal/quick_switch_modal.tsx [hunk 1]
```diff
@@ -157,7 +157,7 @@ export default class QuickSwitchModal extends React.PureComponent<Props, State>
         const providers: SwitchChannelProvider[] = this.channelProviders;
 
         const header = (
-            <h1>
+            <h1 id='quickSwitchHeader'>
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/quick_switch_modal/quick_switch_modal.tsx:160]: curious why we had to do this change?

Comment (saturninoabril) [reply webapp/channels/src/components/quick_switch_modal/quick_switch_modal.tsx:160]: This is the main title of the modal/dialog. Doing so gives the dialog an accessible name by referring to the element that provides the dialog title (by `aria-labelledby='quickSwitchHeader'`). 

Meanwhile, the dialog description (by `aria-describedby='quickSwitchHeaderWithHint'`) will read out as is with title + hint.

Comment (M-ZubairAhmed) [reply webapp/channels/src/components/quick_switch_modal/quick_switch_modal.tsx:160]: 🤦🏽 oh no i didnt checked the aria-labelledby. Sorry

**Pull request merged**